### PR TITLE
Fix surface resizes for android_context completely

### PR DESCRIPTION
**Pull request by sfan5**
Android fucking sucks

tl;dr
\- we need to be able to tell mpv when the surface resized
\- querying these from EGL does not work, it reports outdated values
\- thus we need additionally need those properties to pass the correct dimensions

I agree that my changes can be relicensed to LGPL 2.1 or later.

### CODE REVIEW

**Comment by tmm1** [comment]
Sucks that this is required, but I don't see any better alternative. Patch looks good to me.

The same issue is also described here: https://stackoverflow.com/questions/7603945/eglquerysurface-gives-wrong-size

File: video/out/opengl/context_android.c [hunk 1]
```diff
@@ -26,13 +26,29 @@
...
+
 struct priv {
     struct GL gl;
     EGLDisplay egl_display;
     EGLConfig egl_config;
     EGLContext egl_context;
     EGLSurface egl_surface;
     ANativeWindow *native_window;
+
+    struct android_opts *opts;
```
Comment (wm4) [inline video/out/opengl/context_android.c]: Unused?

**Comment by wm4** [comment]
Regrettable (why is Android so shit?), but LGTM.

File: video/out/opengl/context_android.c (hunk 1 above)
Comment (sfan5) [reply video/out/opengl/context_android.c]: fixed.

**Comment by kevmitch** [comment]
@sfan5 was that just a rebase, or were there any changes?

**Comment by sfan5** [comment]
removed the unused line of code wm4 noted

File: video/out/opengl/context_android.c [hunk 2]
```diff
@@ -122,18 +136,14 @@ static bool android_init(struct ra_ctx *ctx)
...
-    struct priv *p = ctx->priv;
-    int w, h;
+    void *tmp = talloc_new(NULL);
+    struct android_opts *opts = mp_get_config_group(tmp, ctx->global, &android_conf);
 
-    if (!eglQuerySurface(p->egl_display, p->egl_surface, EGL_WIDTH, &w) ||
-        !eglQuerySurface(p->egl_display, p->egl_surface, EGL_HEIGHT, &h)) {
-        MP_FATAL(ctx, "Failed to get height and width!\n");
-        return false;
-    }
```
Comment (tmm1) [inline video/out/opengl/context_android.c]: I think we should fallback to the egl width/height when the user doesn't provide any values.

**Pull request merged**